File: tests/fixtures/code/sample.R
# Summarizes enrollment by wave. Invented for the test suite.
library(dplyr)

# Counts the participants enrolled in one wave.
wave_size <- function(enrollments, wave) {
  nrow(dplyr::filter(enrollments, wave_id == wave))
}

# Returns the waves with no enrollments yet.
empty_waves <- function(enrollments, waves) {
  setdiff(waves, unique(enrollments$wave_id))
}

File: tests/fixtures/code/sample.m
% Summarizes an actigraphy recording, invented for the test suite.
% No real participant data is described here.

function summary = summarize_recording(counts, sampling_hz)
% Returns the mean and the peak of one recording.
    summary.mean = mean(counts);
    summary.peak = max(counts);
    summary.minutes = numel(counts) / (sampling_hz * 60);
end

% Drops the samples recorded while the device was off the wrist.
function kept = drop_offwrist(counts, worn)
    kept = counts(worn);
end
